src/tools/kata-ctl/src/check.rs

```diff
@@ -197,6 +208,110 @@ pub fn check_official_releases() -> Result<()> {
     Ok(())
 }
 
+#[cfg(any(target_arch = "x86_64"))]
+pub fn check_kernel_module_loaded(module: &KernelModule) -> Result<()> {
+    const MODPROBE_PARAMETERS_DRY_RUN: &str = "--dry-run";
+    const MODPROBE_PARAMETERS_FIRST_TIME: &str = "--first-time";
+
+    // Partial check w/ modinfo
+    let output_modinfo = Command::new(MODINFO_PATH)
+        .arg(module.name.clone())
+        .stdout(Stdio::piped())
+        .output()?;
+
+    let status_modinfo = output_modinfo.status;
+
+    // Partial check w/ modprobe
+    let output_modprobe = Command::new(MODPROBE_PATH)
+        .arg(MODPROBE_PARAMETERS_DRY_RUN)
+        .arg(MODPROBE_PARAMETERS_FIRST_TIME)
+        .arg(module.name.clone())
+        .stdout(Stdio::piped())
+        .output()?;
+
+    let status_modprobe = output_modprobe.status;
+
+    // Complete check
+    if status_modinfo.success() && !status_modprobe.success() {
+        return Ok(());
+    }
+
+    match status_modprobe.code() {
+        Some(code) => {
+            // Module not found
+            if code != 0 {
+                let std_err = String::from_utf8_lossy(&output_modinfo.stderr);
+                let msg = format!("{} {}", ERR_LSMOD, std_err);
+                return Err(anyhow!(msg));
+            }
+
+            // module not loaded
+            let msg = format!("{} {}", ERR_MODPROBE, module.name);
+            Err(anyhow!(msg))
+        }
+        None => Err(anyhow!("Process terminated by signal")),
+    }
+}
+
+pub fn check_kernel_module_parameters(
+    module: &KernelModule,
+) -> std::result::Result<std::result::Result<std::string::String, anyhow::Error>, ()> {
```

Comment:
That return type is a bit scary tbh! You could simplify a little (`std::string::String` -> `String`, and maybe `use anyhow::Error`), but it also looks really odd. fwics this is due to `beau_collector`. An alternative would be to simply return a list of results, but wrap it in a `Result` so the caller can check if the function itself worked:

```
```suggestion
pub fn check_kernel_module_parameters(module: &KernelModule) -> Result<Vec<Result<()>> {
```

Reply:
@jodh-intel Thank you, fixed